spyder/plugins/variableexplorer/widgets/collectionseditor.py

```diff
@@ -1115,6 +1123,8 @@ def __init__(self, parent, data, readonly=False, title="",
         self.source_model = CollectionsModelClass(self, data, title,
                                                   names=names,
                                                   minmax=minmax)
+        self.horizontalHeader().sortIndicatorChanged.connect(
+            self.source_model.load_all)

```

Comment:
I don't think this is a good strategy and we should remove it. The thing is CollectionsEditor is used to view the contents of any dictionary, list, tuple and set. We use it as the main widget of the Variable Explorer because Python's namespace is a dictionary, but this change would affect how we display the contents of any other object viewed through it.

So if you have a list with 100,000 elements and you reverse its index, we'd try to show the full contents of it. And that would certainly block Spyder.

Reply:
Ok but we need to know what the new first element is? And what if it is sorted by type or size? how do I know what to show without loading everything?